common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java

```diff
@@ -411,7 +412,13 @@ public boolean doMouseClicked(double mouseX, double mouseY, int button) {
                                     customPoi.getColor());
                         } else {
                             Models.Marker.USER_WAYPOINTS_PROVIDER.addLocation(
-                                    new Location(hovered.getLocation()), iconPoi.getIcon());
+                                    new Location(hovered.getLocation()),
+                                    iconPoi.getIcon(),
+                                    Managers.Feature.getFeatureInstance(WorldWaypointDistanceFeature.class)
+                                                    .showAdditionalTextInWorld
+                                                    .get()
+                                            ? hovered.getName()
+                                            : null);
```

Comment:
Shouldn't this logic be applied in UserWaypointsProvider so it applies to all custom pois?

Reply:
So if I get this right, the right way would be to change all `null` values I put in UserWaypointsProvider to this check?